### PR TITLE
Function to concatenate multiple header files together

### PR DESCRIPTION
**Pull request by gchenfc**
Self-explanatory PR title.

Moves code out of https://github.com/borglab/GTDynamics/pull/130 and into wrap.

### CODE REVIEW

**Comment by varunagrawal** [comment]
Do we need this if we are planning to allow breaking down of interface files?

**Comment by gchenfc** [comment]
Ok,

*** TL;DR - "breaking down of interface files" does this mean also creating different cpp files for pybind as in [here](https://pybind11.readthedocs.io/en/stable/faq.html#how-can-i-reduce-the-build-time) ?  If so, please disregard the rest of this message - I 100% support this.  If you actually mean just splitting up the `.i` files then having pyparsing reassemble them back into one giant `.cpp` file, then I have some comments, but obviously I will take a back seat to whatever you guys have in mind.

1. I will point out one of the reasons I wanted to do this concatenation in cmake was so that cmake flags could allow not compiling certain headers.  Specifically, cablerobot is only compiled when `GTDYNAMICS_BUILD_CABLE_ROBOT=ON` so those wrapper declarations should also _only_ be made when the build flag is on.  I think this could apply to e.g. spider, quadruped, and jumping-robot specific code, and potentially even `gtsam_unstable` which is currently just compiled totally separately from gtsam but could be compiled together according to a compile flag (well there are some advantages to having these in separate wrapper files but in any case...)

2. Another possible advantage is that concatenating the files this way provides more transparent debugging since it is much easier to check the contents of the concatenated .i file rather than inspect into the state of a concatenated string inside python or into the state of a structure of Namespaces or daisy-chained parse rule or something inside python.



As an alternative to point 1 (build flags), we could add some form of #ifdef -ing or otherwise have cmake label an #import to not be used, either through a `configure_file` or passing arguments to the `pybind_wrap.py` script.

As an alternative to point 2 (debugging/introspection), it would be really cool if this actually made separate cpp files which could be compiled separately/in parallel [see here](https://pybind11.readthedocs.io/en/stable/faq.html#how-can-i-reduce-the-build-time).  Perhaps this is really what Varun was getting at - sorry if this was super obvious and it just went over my head.

**Comment by ProfFan** [comment]
Note: we can already do multi-file, provided that they are all imported by the top-level Python module. e.g. generating multiple binaries like `gtdynamics_core.cpythonxxx.so`, `gtdynamics_utils.cpythonxx.so`, etc etc. and in top level we do
```
core.py:
from gtdynamics_core import *

utils.py:
from gtdynamics_utils import *
```

**Comment by varunagrawal** [comment]
> Note: we can already do multi-file, provided that they are all imported by the top-level Python module. e.g. generating multiple binaries like `gtdynamics_core.cpythonxxx.so`, `gtdynamics_utils.cpythonxx.so`, etc etc. and in top level we do
> ```
> core.py:
> from gtdynamics_core import *
> 
> utils.py:
> from gtdynamics_utils import *
> ```

This needs to go into wrap's docs with more detail.

**Comment by varunagrawal** [comment]
Umm I am confused, this PR does not do what the title says and the PR comment is consequently not helpful.

**Comment by gchenfc** [comment]
Sorry my mistake, I meant to revert everything except for the very first commit, but I accidentally reverted only the very first commit!!!

One sec

**Comment by gchenfc** [comment]
@varunagrawal as per Frank's comment earlier, since breaking down of interface files might take a while, can we merge this in so that https://github.com/borglab/GTDynamics/pull/130 can be merged?

Then we can fix this once multiple interface files is done.

**Review by dellaert — APPROVED** [review]

**Comment by varunagrawal** [comment]
This should have been squashed and merged due to all the reverts, but okay.
